src/Geocoder/Provider/GoogleMaps.php

```diff
@@ -153,8 +153,9 @@ private function executeQuery($query)
             throw new NoResult(sprintf('Could not execute query "%s".', $query));
         }
 
-        if ('REQUEST_DENIED' === $json->status && 'The provided API key is invalid.' === $json->error_message) {
-            throw new InvalidCredentials(sprintf('API key is invalid %s', $query));
+        if ('REQUEST_DENIED' === $json->status) {
```

Comment:
There are different messages with that code. I for example got a message that access from my IP is not allowed on the API key and the IP that google recognizes is dumped. So it is very helpful to include such messages in the exception.